common/blobstore/blob/key_test.go

```diff
@@ -149,6 +155,13 @@ func (s *KeySuite) TestNewKeyFromString() {
 			expectExtension:  "ext",
 			expectNamePieces: []string{"foo1", "bar2", "3baz"},
 		},
+		{
+			inputStr:         "foo1_bar2_-10.ext",
+			expectError:      false,
+			expectBuiltKey:   "foo1_bar2_-10.ext",
```

Comment:
Any reason to allow `-`.. Looks a little odd.

Reply:
Negative numbers for close version in case of single DC